src/frequenz/client/microgrid/_id.py

```diff
@@ -105,3 +105,53 @@ def __repr__(self) -> str:
     def __str__(self) -> str:
         """Return the short string representation of this instance."""
         return f"CID{self._id}"
+
+
+@final
+class SensorId:
+    """A unique identifier for a microgrid sensor."""
+
+    def __init__(self, id_: int, /) -> None:
+        """Initialize this instance.
+
+        Args:
+            id_: The numeric unique identifier of the microgrid sensor.
+
+        Raises:
+            ValueError: If the ID is negative.
+        """
+        if id_ < 0:
+            raise ValueError("Sensor ID can't be negative.")
+        self._id = id_
+
+    def __int__(self) -> int:
+        """Return the numeric ID of this instance."""
+        return self._id
+
+    def __eq__(self, other: object) -> bool:
+        """Check if this instance is equal to another object."""
+        # This is not an unidiomatic typecheck, that's an odd name for the check.
```

Comment:
I feel like whether it is odd or not is out of scope for this comment.

```suggestion
        # This is not an unidiomatic typecheck, pylint is wrong.

Reply:
Fair, not convinced about the plain **wrong** here, but I'll see if I can put it in a more relevant way. Although this is in every ID class, so I would have to update all comments.

Reply:
It says it is unidiomatic, you say it is not, so it is wrong.  But more context sounds good, of course.

Reply:
**THANK YOU**. I had to redo the whole module because of this :stuck_out_tongue_closed_eyes: